client/pages/home.jsx

```diff
@@ -1,8 +1,194 @@
+  },
+  icons: {
+    width: '100%',
+    height: '100px'
+  }
+};
+
+export default class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.folders = null;
+    this.batches = null;
+    this.maxFolders = 5;
+    this.maxBatches = 6;
+    this.createNew = null;
+    this.trash = null;
+    this.state = {
+      folders: [],
+      batches: [],
+      openedId: 0
+    };
+
+    this.displayFolders = this.displayFolders.bind(this);
+    this.displayCards = this.displayCards.bind(this);
+    this.addNewFolder = this.addNewFolder.bind(this);
+    this.clickFolder = this.clickFolder.bind(this);
+    this.recentBatch = this.recentBatch.bind(this);
+    this.trashFolder = this.trashFolder.bind(this);
+  }
+
+  componentDidMount() {
+    this.displayFolders();
+  }
+
+  displayFolders() {
+    fetch('/api/folders')
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          folders: data
+        });
+      })
+      .catch(err => console.error('Fetch folders failed!', err));
+  }
+
+  displayCards(folderId) {
+    fetch(`/api/batches/${folderId}`)
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          batches: data
+        });
+      })
+      .catch(err => console.error('Fetch failed!', err));
+  }
+
+  addNewFolder() {
+    if (this.folders.length >= this.maxFolders) return;
+    const count = this.folders.length + 1;
+    const folder = {
+      folderName: 'Folder ' + count,
+      userId: 1
+    };
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(folder)
+    };
+    fetch('/api/folders/', req)
+      .then(res => res.json())
+      .then(result => { })
+      .catch(error => console.error('Post folder error!', error));
+    this.displayFolders();
+  }
+
+  clickFolder(folderId) {
+    this.setState({ openedId: folderId });
+    this.displayCards(folderId);
+    this.createNew = <img className="create-new-flashcards"
+      onClick={() => this.props.setActiveFolder(this.state.openedId)}></img>;
+    this.trash = <img className="trash" onClick={() => this.trashFolder(this.state.openedId)}></img>;
+  }
+
+  recentBatch() {
+    let batch = null;
+    fetch('/api/batches/')
+      .then(res => res.json())
+      .then(result => {
+        let biggest; let bigIndex = 0;
+        for (let i = 0; i < result.length; i++) {
+          if (result[i].batchId > biggest) {
+            biggest = result[i].batchId;
+            bigIndex = i;
+          }
+        }
+        batch = result[bigIndex];
+      })
+      .catch(error => console.error('recentBatch failed!', error));
+    this.props.setActiveBatch(batch);
+  }
+
+  trashFolder(folderId) {
+    const req = {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+    fetch(`api/folders/${folderId}`, req)
+      .then(res => res.json())
```

Comment:
Because your delete endpoint sends no data back when you call `res.json()` it will throw an error because it can't parse nothing.